src/frontend/src/components/SearchPage/SearchPage.jsx

```diff
@@ -62,38 +67,74 @@ const SearchPage = () => {
             },
           };
         });
-      setResults(finalResults);
+      setResults({ type: 'author', searchResults: finalResults });
     },
   });
 
   // Hook that will re-render the page only if the state of searchText changes
   useEffect(() => {
-    executeSearch();
-  }, [searchText]);
+    async function searchPosts() {
+      const res = await fetch(`${telescopeUrl}/query?search=${searchText}`);
+      const searchResults = await res.json();
+      const postIds = searchResults.values.map((result) => result.id);
+      let posts = await Promise.all(postIds.map((id) => fetch(`${telescopeUrl}/posts/${id}`)));
+      posts = await Promise.all(posts.map((post) => post.json()));
+      setResults({ type: 'post', searchResults: posts });
+    }
+
+    if (filter === 'author') executeSearch();
+    else if (filter === 'post') searchPosts();
+  }, [searchText, filter, telescopeUrl, executeSearch]);
 
   // Displays one of three options depending on whether there is a search string, results and no results
   const displayResults = () => {
-    if (searchText.length === 0 && results.length < 1) {
+    if (searchText.length > 0 && loading) {
+      return (
+        <h1 className={classes.searchReply}>
```

Comment:
We need better CSS on this to at least get it in the centre

Reply:
It is now in the center